web/profiles/joinup/src/Form/UserCancelForm.php

```diff
@@ -0,0 +1,113 @@
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Overrides the default implementation of the form to delete a user account.
+ *
+ * Deletion of a user account will be denied when the user is the sole owner of
+ * a collection.
+ */
+class UserCancelForm extends CoreUserCancelForm {
+
+  /**
+   * The relation manager service.
+   *
+   * @var \Drupal\joinup_core\JoinupRelationManager
+   */
+  protected $relationManager;
+
+  /**
+   * Instantiates a new UserCancelForm class.
+   *
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   *   The entity manager.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entity_type_bundle_info
+   *   The entity type bundle service.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   The time service.
+   * @param \Drupal\joinup_core\JoinupRelationManager $relation_manager
+   *   The Joinup relation manager.
+   */
+  public function __construct(EntityManagerInterface $entity_manager, EntityTypeBundleInfoInterface $entity_type_bundle_info = NULL, TimeInterface $time = NULL, JoinupRelationManager $relation_manager = NULL) {
+    parent::__construct($entity_manager, $entity_type_bundle_info, $time);
+
+    // Replicate the behaviour of the parent implementation.
+    $this->relationManager = $relation_manager ?: \Drupal::service('joinup_core.relations_manager');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity.manager'),
+      $container->get('entity_type.bundle.info'),
+      $container->get('datetime.time'),
+      $container->get('joinup_core.relations_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    // Prepare a list of collections where the user is the sole owner.
+    $collections = $this->relationManager->getCollectionsWhereSoleOwner($this->entity);
+
+    if (!empty($collections)) {
+      $form = [
+        'collections' => [
+          '#theme' => 'item_list',
+          '#items' => array_map(function (EntityInterface $collection) {
+            return $collection->toLink($collection->label());
+          }, $collections),
+          '#weight' => 0,
+        ],
+        'actions' => [
+          '#type' => 'actions',
+          // @see \Drupal\Core\Entity\ContentEntityConfirmFormBase::actions()
+          'cancel' => ConfirmFormHelper::buildCancelLink($this, $this->getRequest()),
+          '#weight' => 10,
+        ],
+      ];
+
+      // Show a nicer message when the user is trying to delete its own account.
+      if ($this->currentUser()->id() === $this->entity->id()) {
+        $form['pre_warning'] = [
+          '#markup' => $this->t('Dear @name,<br />when processing your request to delete your account, we noticed that you are the sole owner of these collections:', [
```

Comment:
That was implemented that way because we converted a notification message into a confirmation message (because the other way around would be overkill).